service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/NonConsultationResourceMapper.java

```diff
@@ -136,9 +139,26 @@ private EncounterTemplateParametersBuilder buildForReferralRequest(String compon
             .altCode(DEFAULT_CODE);
     }
 
-    // TODO: Add builder once NIAD-910 has been completed
-    private EncounterTemplateParametersBuilder buildForDiagnosticRequest(String component) {
-        return null;
+    private EncounterTemplateParametersBuilder buildForDiagnosticReport(String component, DiagnosticReport diagnosticReport) {
+        boolean isAgentPerson = false;
+
+        if (diagnosticReport.hasPerformer()) {
+            var agent = Optional.ofNullable(diagnosticReport.getPerformerFirstRep().getActor());
+            if (agent.isPresent()) {
+                isAgentPerson = agent.get().getReferenceElement().getResourceType().equals(ResourceType.Practitioner.name());
+            }
+        }
```

Comment:
This could be combined
```suggestion
        if (diagnosticReport.hasPerformer()) {
            isAgentPerson = Optional.of(diagnosticReport.getPerformerFirstRep())
                .map(DiagnosticReport.DiagnosticReportPerformerComponent::getActor)
                .map(Reference::getReferenceElement)
                .filter(IIdType::hasResourceType)
                .filter(idType -> idType.getResourceType().equals(ResourceType.Practitioner.name()))
                .isPresent();
        }
```